mailjet_simple.module

```diff
@@ -184,6 +184,28 @@ function mailjet_simple_user_register_form_submit($form, &$form_state) {
 
     // Create the contact object in Mailjet.
     $contact = $api->updateContact($values, $lists);
+
+    // Delete old contact objects in Mailjet email address is changed.
+    $new_mail = $form_state['values']['mail'];
+    $old_mail = $form['account']['mail']['#default_value'];
+
+    if ($new_mail !== $old_mail) {
+      $unsubscribe_lists = [];
+      $unsubscribe_values = array(
+        'Email' => $old_mail,
+      );
+
+      foreach ($form_state['values']['mailjet_simple']['contact_lists'] as $list_id => $selected) {
```

Comment:
Use mailjet_simple_get_list_ids_by_mail($mail) to build an array of list ids for sub/unsub